Framework/Core/include/Framework/InputRecord.h

```diff
@@ -516,6 +523,7 @@ class InputRecord
  private:
   std::vector<InputRoute> const& mInputsSchema;
   InputSpan mSpan;
+  std::vector<std::unique_ptr<boost::container::pmr::memory_resource>> mSpectatorResources;
```

Comment:
Maybe we should ask that FairMQMessage becomes a "spectator resource" itself, so that we do not need to have separate accounting of those. @rbx @mkrzewic what do you think?

Reply:
Nevermind the FairMQ message does not know the type...

Reply:
We have been playing with some ideas at FairMQLevel (see e.g. https://github.com/FairRootGroup/FairMQ/pull/110). btw, the resource does not need to know the type, it is meant to allocate bytes only.

Reply:
i mean not only, but bytes is *enough* :)

Reply:
Ah, I missed resourcebuffer was just "char".